src/main/cpp/fielddata/BargeHelper.cpp

```diff
@@ -0,0 +1,119 @@
+
+//====================================================================================================================================================
+// Copyright 2025 Lake Orion Robotics FIRST Team 302
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE
+// OR OTHER DEALINGS IN THE SOFTWARE.
+//====================================================================================================================================================
+
+#include "chassis/definitions/ChassisConfig.h"
+#include "chassis/definitions/ChassisConfigMgr.h"
+#include "fielddata/BargeHelper.h"
+#include "frc/DriverStation.h"
+#include "utils/FMSData.h"
+#include "utils/logging/debug/Logger.h"
+#include "frc/Filesystem.h"
+
+BargeHelper *BargeHelper::m_instance = nullptr;
+BargeHelper *BargeHelper::GetInstance()
+{
+    if (BargeHelper::m_instance == nullptr)
+    {
+        BargeHelper::m_instance = new BargeHelper();
+    }
+    return BargeHelper::m_instance;
+}
+
+BargeHelper::BargeHelper() : m_chassis(ChassisConfigMgr::GetInstance()->GetCurrentChassis()),
+                             m_allianceColor(FMSData::GetInstance()->GetAllianceColor()),
```

Comment:
This may be getting created before the alliance color is set; some of the logic inside this file may need to change as a result